src/components/Activity/FilterList/FilterList.tsx

```diff
@@ -0,0 +1,70 @@
+import Icon from "src/components/ui/Icon";
+import { Filter } from "src/types/Activity";
+
+import style from './FilterList.module.css'
+
+interface FilterListProps {
+    filters: Filter; 
+    currentFilter:string;
+    onChangeFilter:(value:string)=>void
+}
+
+const FilterList = ({filters, onChangeFilter, currentFilter}:FilterListProps) => {
+ const {all, completed, todo, progress} = filters
+    return ( <ul className={style.filterList}>
+        <li
+          className={style.filterItem}
+          onClick={() => onChangeFilter(all.value)}
+        >
+          <Icon
+            id={
+              currentFilter === all.value
+                ? 'icon-checkbox_true'
+                : 'icon-checkbox_false'
+            }
+          />
+          <span>{filters.all.lable}</span>
+        </li>
+        <li
+          className={style.filterItem}
+          onClick={() => onChangeFilter(completed.value)}
+        >
+          <Icon
+            id={
+              currentFilter === completed.value
+                ? 'icon-checkbox_true'
+                : 'icon-checkbox_false'
+            }
+          />
+          <span>{completed.lable}</span>
+        </li>
+        <li
+          className={style.filterItem}
+          onClick={() => onChangeFilter(todo.value)}
+        >
+          <Icon
+            id={
+              currentFilter === todo.value
+                ? 'icon-checkbox_true'
+                : 'icon-checkbox_false'
+            }
+          />
+          <span>{todo.lable}</span>
+        </li>
+        <li
+          className={style.filterItem}
+          onClick={() => onChangeFilter(progress.value)}
+        >
+          <Icon
+            id={
+              currentFilter === progress.value
+                ? 'icon-checkbox_true'
+                : 'icon-checkbox_false'
+            }
+          />
+          <span>{progress.lable}</span>
+        </li>
+      </ul> );
+}
```

Comment:
there are a lot of duplicated layout, can you try to exclude this into array and below map() it?